clustermesh/clustermesh.go

```diff
@@ -990,9 +994,9 @@ func (k *K8sClusterMesh) Disconnect(ctx context.Context) error {
 }
 
 type Status struct {
-	AccessInformation *accessInformation
-	Service           *corev1.Service
-	Connectivity      *ConnectivityStatus
+	AccessInformation *accessInformation  `json:"access_information,omitempty"`
+	Service           *corev1.Service     `json:"service,omitempty"`
```

Comment:
corev1.Service is kind of verbose with all the fields such as managedFields, etc, so normally, we are using cilium slim type instead.

Please find below the patch if you want to address this point

<details>
<summary>diff</summary>

```diff
diff --git a/clustermesh/clustermesh.go b/clustermesh/clustermesh.go
index 63c70380..b2d6f8ba 100644
--- a/clustermesh/clustermesh.go
+++ b/clustermesh/clustermesh.go
@@ -25,7 +25,9 @@ import (
    "k8s.io/apimachinery/pkg/util/intstr"
 
    "github.com/cilium/cilium/api/v1/models"
+   ciliumK8s "github.com/cilium/cilium/pkg/k8s"
    ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+   slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
 
    "github.com/cilium/cilium-cli/defaults"
    "github.com/cilium/cilium-cli/internal/certs"
@@ -1037,9 +1039,9 @@ func (k *K8sClusterMesh) Disconnect(ctx context.Context) error {
 }
 
 type Status struct {
-   AccessInformation *accessInformation  `json:"access_information,omitempty"`
-   Service           *corev1.Service     `json:"service,omitempty"`
-   Connectivity      *ConnectivityStatus `json:"connectivity,omitempty"`
+   AccessInformation *accessInformation   `json:"access_information,omitempty"`
+   Service           *slim_corev1.Service `json:"service,omitempty"`
+   Connectivity      *ConnectivityStatus  `json:"connectivity,omitempty"`
 }
 
 func (k *K8sClusterMesh) statusAccessInformation(ctx context.Context, log bool, getExternalWorkloadSecret bool) (*accessInformation, error) {
@@ -1062,7 +1064,7 @@ retry:
    return ai, err
 }
 
-func (k *K8sClusterMesh) statusService(ctx context.Context) (*corev1.Service, error) {
+func (k *K8sClusterMesh) statusService(ctx context.Context) (*slim_corev1.Service, error) {
    w := utils.NewWaitObserver(ctx, utils.WaitParameters{Log: func(err error, wait string) {
        k.Log("⌛ Waiting (%s) for ClusterMesh service to be available: %s", wait, err)
    }})
@@ -1081,7 +1083,12 @@ retry:
        return nil, fmt.Errorf("clustermesh-apiserver cannot be found: %w", err)
    }
 
-   return svc, nil
+   r, ok := ciliumK8s.ConvertToK8sService(svc).(*slim_corev1.Service)
+   if !ok {
+       return nil, fmt.Errorf("unable to convert service %q to slim_corev1.Service", svc.Name)
+   }
+
+   return r, nil
 }
 
 func (k *K8sClusterMesh) waitForDeployment(ctx context.Context) error {
@@ -1275,7 +1282,7 @@ func (k *K8sClusterMesh) Status(ctx context.Context) (*Status, error) {
 
    k.Log("✅ Service %q of type %q found", defaults.ClusterMeshServiceName, s.Service.Spec.Type)
 
-   if s.Service.Spec.Type == corev1.ServiceTypeLoadBalancer {
+   if s.Service.Spec.Type == slim_corev1.ServiceTypeLoadBalancer {
        if len(s.AccessInformation.ServiceIPs) == 0 {
            k.Log("❌ Service is of type LoadBalancer but has no IPs assigned")
            return nil, fmt.Errorf("no IP available to reach cluster")
@@ -1687,7 +1694,7 @@ func (k *K8sClusterMesh) ExternalWorkloadStatus(ctx context.Context, names []str
 
    k.Log("✅ Service %q of type %q found", defaults.ClusterMeshServiceName, svc.Spec.Type)
 
-   if svc.Spec.Type == corev1.ServiceTypeLoadBalancer {
+   if svc.Spec.Type == slim_corev1.ServiceTypeLoadBalancer {
        if len(ai.ServiceIPs) == 0 {
            k.Log("❌ Service is of type LoadBalancer but has no IPs assigned")
            return fmt.Errorf("no IP available to reach cluster")
```

</details>

Reply:
I'm fine with using `slim_core` @sayboras . That means we need to bump the `cilium/cilium` vendoring though. Should I add this to this PR?